packages/world/ts/config/resolveWorldConfig.ts

```diff
@@ -12,12 +12,14 @@ export type ResolvedWorldConfig = ReturnType<typeof resolveWorldConfig>;
  * splitting the access list into addresses and system names.
  */
 export function resolveWorldConfig(
-  config: Pick<StoreConfig & WorldConfig, "systems" | "excludeSystems">,
+  config: Pick<StoreConfig & WorldConfig, "namespace" | "systems" | "excludeSystems">,
   existingContracts?: string[],
 ) {
   // Include contract names ending in "System", but not the base "System" contract, and not Interfaces
   const defaultSystemNames =
-    existingContracts?.filter((name) => name.endsWith("System") && name !== "System" && !name.match(/^I[A-Z]/)) ?? [];
+    existingContracts?.filter(
+      (name) => name.endsWith("System") && name !== "System" && !name.match(/^I[A-Z]/) && !name.match(/__I/),
```

Comment:
no change needed here, but just FYI, `.match` returns data where as `.test` just returns a boolean, so I tend to use it more often for conditionals like this

```ts
/__I/.test(name)
```

Reply:
not sure if you find this more or less readable but could combine the regex test and also ensure the `__I` variant matches the `[A-Z]` constraint
```suggestion
      (name) => name.endsWith("System") && name !== "System" && !/(^|__)I[A-Z]/.test(name),
```